gradle/validation/rat-sources.gradle

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import groovy.xml.NamespaceBuilder
+
+allprojects {
+    configurations {
+        rat
+    }
+
+    dependencies {
+        rat 'org.apache.rat:apache-rat-tasks:0.13'
+    }
+}
+
+subprojects {
+    plugins.withId("java", {
+        task("rat", type: RatTask) {
+            group = 'Verification'
+            description = 'Runs Apache Rat checks.'
+        }
+    })
+}
+
+// Somewhat inspired by existing task from Apache Kafka
+class RatTask extends DefaultTask {
+    @Input
+    List<String> includes = []
+
+    @Input
+    List<String> excludes = []
+
+    @Input
+    List<String> additionalExcludes = []
+
+    def reportDir = new File(project.buildDir, 'rat')
+    def xmlReport = new File(reportDir, 'rat-report.xml')
+
+    def generateXmlReport(File reportDir) {
+        def uri = 'antlib:org.apache.rat.anttasks'
+        def ratClasspath = project.configurations.rat.asPath
+        ant.taskdef(resource: 'org/apache/rat/anttasks/antlib.xml', uri: uri, classpath: ratClasspath)
+
+        def projectPath = project.projectDir.absolutePath
+
+        def rat = NamespaceBuilder.newInstance(ant, uri)
+        rat.report(format: 'xml', reportFile: xmlReport, addDefaultLicenseMatchers: true) {
+            ant.fileset(dir: projectPath) {
+                ant.include(name: '*.xml')
+                includes.each {
+                    ant.include(name: it)
+                }
+                additionalExcludes.each {
+                    ant.exclude(name: it)
+                }
+            }
+
+            // There may be a more efficient way to write these next two blocks since they use the same logic
+            project.sourceSets.main.java.srcDirs.each {
+                ant.fileset(dir: it, erroronmissingdir: false) {
```

Comment:
I'd be explicit and use variable for closures rather than "it", especially in larger code blocks.